src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/AddStandardTrainingLocationControllerTests/AddStandardTrainingLocationControllerPostTests.cs

```diff
@@ -0,0 +1,89 @@
+            sessionServiceMock.Setup(s => s.Get<StandardSessionModel>(It.IsAny<string>()))
+                .Returns((StandardSessionModel)null);
+
+            var result = await sut.SubmitAProviderlocation(new CourseLocationAddSubmitModel());
+
+            result.As<RedirectToRouteResult>().Should().NotBeNull();
+            result.As<RedirectToRouteResult>().RouteName.Should().Be(RouteNames.GetAddStandardSelectStandard);
+        }
+
+        [Test, MoqAutoData]
+        public async Task Get_ModelStateIsInvalid_ReturnsViewResult(
+            [Frozen] Mock<ISessionService> sessionServiceMock,
+            [Greedy] AddStandardTrainingLocationController sut,
+            StandardSessionModel standardSessionModel,
+            string cancelLink)
+        {
+            sut.AddDefaultContextWithUser().AddUrlHelperMock().AddUrlForRoute(RouteNames.GetNewStandardViewTrainingLocationOptions, cancelLink);
+            sessionServiceMock.Setup(s => s.Get<StandardSessionModel>(It.IsAny<string>()))
+                .Returns(standardSessionModel);
+            sut.ModelState.AddModelError("key", "message");
+
+            var result = await sut.SubmitAProviderlocation(new CourseLocationAddSubmitModel());
+
+            result.As<ViewResult>().Should().NotBeNull();
+            result.As<ViewResult>().ViewName.Should().Be(AddStandardTrainingLocationController.ViewPath);
+            result.As<ViewResult>().Model.As<CourseLocationAddViewModel>().CancelLink.Should().Be(cancelLink);
+        }
+
+        [Test, MoqAutoData]
+        public async Task Get_ModelStateIsValid_UpdatesStandardSessionModel(
+            [Frozen] Mock<IMediator> mediatorMock,
+            [Frozen] Mock<ISessionService> sessionServiceMock,
+            [Greedy] AddStandardTrainingLocationController sut,
+            CourseLocationAddViewModel submitModel,
+            string cancelLink)
+        {
+            var navigationId = Guid.NewGuid();
+            var allLocations = new GetAllProviderLocationsQueryResult
+            {
+                ProviderLocations = new List<ProviderLocation>
+                    { new ProviderLocation { LocationType = LocationType.Provider, LocationName = "location name",NavigationId = navigationId} }
+            };
+
+            submitModel.TrainingVenueNavigationId = navigationId.ToString();
+            sut.AddDefaultContextWithUser().AddUrlHelperMock().AddUrlForRoute(RouteNames.GetNewStandardViewTrainingLocationOptions, cancelLink);
+            mediatorMock.Setup(m => m.Send(It.IsAny<GetAllProviderLocationsQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(allLocations);
+            sessionServiceMock.Setup(s => s.Get<StandardSessionModel>(It.IsAny<string>())).Returns(new StandardSessionModel { LarsCode = larsCode });
+
+            var result = await sut.SubmitAProviderlocation(submitModel);
+
+            sessionServiceMock.Verify(s => s.Set(It.IsAny<StandardSessionModel>(), TestConstants.DefaultUkprn));
```

Comment:
This verify statement should check that the location has been added to the model with right values